### ISSUE TITLE
`no-trailing-spaces `: `markdown` variant

### ISSUE THREAD

**Issue by Dimava** [eslint-stylistic/eslint-stylistic#397]
### Clear and concise description of the problem

`.md` files and comments often use "markdown newline" which is two spaces on the end of line

### Suggested solution

Please add a "markdown" option that only allows two spaces (`  `) on the end of line not followed by an empty line

### Additional context

`·` means a space here
Good code:
```ts
// file.js
/** 
 * first line··
 * second line
 */
let foo = 0;
```

```md
<!-- file.md -->
first line··
second line
``` 

Bad code: 

```ts
// file.js
/** 
 * first line··· // three+ spaces is bad, should be 2
 * second line· // one space is bad, should be 0
 * third line line↹ // tab is bad
 * fourth line·· // spaces before empty line is bad
 *
 * fifth line·· // spaces before end is bad
 */
let foo = 0;
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
> .md files and comments often use "markdown newline" which is two spaces on the end of line

Can you explain more why and how it works? Isn't that having trailing space could be confusing and quite implicit?

**Comment by Dimava** [comment]
![image](https://github.com/eslint-stylistic/eslint-stylistic/assets/20068737/a061eadb-ca10-44e5-84dc-032d02bed2ca)

**Comment by Dimava** [comment]
Hmm, apparently you can also use `\` and `<br>` for markdown
```
first line··
second line\
third line<br>
fourth line
```
`<br>` doesn't work in VSCode JSDoc tho
